packages/components/src/site-thumbnail/use-mshots-img.tsx

```diff
@@ -24,98 +23,55 @@ export type MShotsOptions = {
 	requeue?: boolean;
 };
 
-// This custom react hook returns undefined while the image is loading and
-// a HTMLImageElement (i.e. the class you get from `new Image()`) once loading
-// is complete.
-//
-// It also triggers a re-render (via setState()) when the value changes, so just
-// check if it's truthy and then treat it like any other Image.
-//
-// Note the loading may occur immediately and synchronously if the image is
-// already or may take up to several seconds if mshots has to generate and cache
-// new images.
-//
-// The calling code doesn't need to worry about the details except that you'll
-// want some sort of loading display.
-//
-// Inspired by https://stackoverflow.com/a/60458593
 export const useMshotsImg = (
 	src: string,
 	options: MShotsOptions
-): HTMLImageElement | undefined => {
-	const [ loadedImg, setLoadedImg ] = useState< HTMLImageElement >();
-	const [ count, setCount ] = useState( 0 );
-	const previousSrc = useRef( src );
-
-	const imgRef = useRef< HTMLImageElement >();
-	const timeoutIdRef = useRef< number >();
-
-	const previousImg = useRef< HTMLImageElement >();
-	const previousOptions = useRef< MShotsOptions >();
-	// Oddly, we need to assign to current here after ref creation in order to
-	// pass the equivalence check and avoid a spurious reset
-	previousOptions.current = options;
+): {
+	src: string;
+	isLoading: boolean;
+	isError: boolean;
+	imgRef: React.MutableRefObject< HTMLImageElement | null >;
+} => {
+	const [ retryCount, setRetryCount ] = useState( 0 );
+	const mshotUrl = useMemo(
```

Comment:
Out of curiosity, what's the significance of this use of `useMemo()`?

Reply:
There are two main reasons to use `useMemo`.
1. The computation is heavy and we want to cache the result
2. We are assigning an object, passing it as a dependency, and we don't want to recreate the object in each render, because it would trigger the execution of the `useEffect`.

In this example, `mshotUrl` is a literal (string), so we don't strictly need to add useMemo. But it feels correct to me, and more resilient to future changes.

Reply:
👍 Makes sense, thanks for the explanation!